modules/consul/instance.tf

```diff
@@ -0,0 +1,83 @@
+resource "aws_key_pair" "consul_cluster" {
+  key_name   = "consul_cluster"
+  public_key = ""
+}
+
+resource "aws_instance" "consul_server" {
+  count             = "${var.count_consul}"
+  instance_type     = "${var.instance_type}"
+  ami               = "${var.ami}"
+  availability_zone = "${var.availability_zone}"
+  subnet_id         = "${var.vpc_network[0]}"
+  security_groups   = ["${var.security_group}"]
+  key_name          = "${aws_key_pair.consul_cluster.id}"
+
+  tags = {
+    Name = "consul-server-${count.index}"
+  }
+}
+
+resource "null_resource" consul_cluster {
+  depends_on = ["aws_instance.consul_server"]
+  count      = 2
+
+  connection {
+    host        = "${element(aws_instance.consul_server.*.public_ip, count.index)}"
+    type        = "ssh"
+    user        = "ubuntu"
+    private_key = "${file("${var.aws_pem_key_file_path}")}"
+  }
+
+  provisioner "file" {
+    source      = "keys/${count.index}"
+    destination = "/home/ubuntu/keys"
+  }
+
+  provisioner "file" {
+    source      = "keys/ca.pem"
+    destination = "/home/ubuntu/keys/ca.pem"
+  }
+
+  provisioner "file" {
+    source      = "vault"
+    destination = "/home/ubuntu/vault"
+  }
+
+  provisioner "file" {
+    source      = "modules/consul/templates/config.json"
+    destination = "/home/ubuntu/config.json"
+  }
+
+  provisioner "file" {
+    content     = "${element(data.template_file.vault_conf.*.rendered, count.index)}"
+    destination = "/home/ubuntu/vault/config/local.json"
+  }
+
+  provisioner "remote-exec" {
```

Comment:
In this particular case, it's better to use templates and conduct provision script in a separate file. In Production better to use SCM.